schemas/component/ui.schema

```diff
@@ -1,113 +1,189 @@
+                    "items": {
+                        "type": "string"
+                    },
+                    "examples": [
+                        [
+                            "*"
+                        ]
+                    ]
+                },
+                "placeholder": {
+                    "title": "Placeholder",
+                    "description": "Placeholder override. Default is `examples` property in the schema.",
+                    "type": "string"
+                },
+                "properties": {
+                    "title": "Properties",
+                    "description": "A map of component property names to UI options with customizations for each property.",
+                    "type": "object",
+                    "additionalProperties": {
+                        "title": "Property Name",
+                        "description": "UI Options object describing UI customizations for this property.",
+                        "$ref": "#/definitions/FormUIOptions"
+                    }
+                },
+                "subtitle": {
+                    "title": "Subtitle",
+                    "description": "Subtitle rendered in form title, defaults to schema.$kind",
+                    "type": "string"
+                },
+                "widget": {
+                    "title": "Field Widget",
+                    "description": "Override default field widget.",
+                    "type": "string",
+                    "enum": [
+                        "checkbox",
+                        "date",
+                        "datetime",
+                        "input",
+                        "number",
+                        "radio",
+                        "select",
+                        "textarea"
+                    ]
+                }
+            }
+        },
+        "MenuOptions": {
+            "type": "object",
+            "properties": {
+                "label": {
+                    "title": "Menu Item Label",
+                    "description": "Text that appears as the menu item. Defaults to $kind.",
+                    "type": "string"
+                },
+                "submenu": {
+                    "title": "Submenu",
+                    "description": "An array of menu item labels which get mapped to a menu hierarchy. Set to false to leave the component un-grouped. To have a component appear in multiple submenus, provide multiple arrays.",
+                    "oneOf": [
+                        {
+                            "$ref": "#/definitions/SubmenuOptions"
+                        },
+                        {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/SubmenuOptions"
+                            }
+                        },
+                        {
+                            "const": false
+                        }
```

Comment:
Does `"const": false` need to be included in the `oneOf`? We can assume the submenu property is falsy if it's undefined.

Reply:
I think if this is undefined, we would group the action in the Custom Actions menu item. But that's a good question. @yeze322 do you have any thoughts?

Reply:
When "label" is `undefined` or `''`, will the default label resolver generate a fallback label string?
If so, I think `false` is better to indicate that user whats to hide the label.

Reply:
We do use false elsewhere to indicate hiding the label. In this case, there is no option to hide the label. We could use `null` instead. What are your thoughts?